app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+
+    IconButton(
+        onClick = {
+          // TODO : Adapt the map screen to navigation
+          Toast.makeText(context, "Menu not implemented yet", Toast.LENGTH_SHORT).show()
+        },
+        modifier =
+            Modifier.padding(16.dp)
+                .align(Alignment.TopStart)
+                // Clip needs to be before background
+                .clip(RoundedCornerShape(8.dp))
+                .background(MaterialTheme.colorScheme.surface)
+                .testTag(MapScreen.TEST_TAG_MENU_BUTTON)) {
+          Icon(
+              Icons.Default.Menu,
+              contentDescription =
+                  context.getString(R.string.map_screen_menu_button_content_description),
+              tint = MaterialTheme.colorScheme.onSurface)
+        }
+
+    CollapsibleHikesList(hikingRoutesViewModel)
+  }
+
+  // Initialize the list of hiking routes once when the map is loaded
+  // Leave subsequent updates to the map listener
+  LaunchedEffect(Unit) { hikingRoutesViewModel.setArea(mapView.boundingBox) }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CollapsibleHikesList(hikingRoutesViewModel: ListOfHikeRoutesViewModel) {
+  val scaffoldState = rememberBottomSheetScaffoldState()
+  val routes = hikingRoutesViewModel.hikeRoutes.collectAsState()
+  val context = LocalContext.current
+
+  BottomSheetScaffold(
+      scaffoldState = scaffoldState,
+      sheetContainerColor = MaterialTheme.colorScheme.surface,
+      sheetContent = {
+        Column(modifier = Modifier.fillMaxSize().testTag(MapScreen.TEST_TAG_HIKES_LIST)) {
+          LazyColumn(modifier = Modifier.fillMaxSize()) {
+            if (routes.value.isEmpty()) {
+              item {
+                // Use a box to center the Text composable of the empty list message
+                Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
+                  Text(
+                      text = context.getString(R.string.map_screen_empty_hikes_list_message),
+                      style = MaterialTheme.typography.bodyLarge,
+                      // Align the text within the Text composable to the center
+                      textAlign = TextAlign.Center)
+                }
+              }
+            } else {
+              items(routes.value.size) { index: Int ->
+                HikingRouteItem(
+                    title = routes.value[index],
+                    altitudeDifference = 1000,
+                    isSuitable = index % 2 == 0,
+                    onClick = {
+                      // TODO : Navigate to the details of the selected hiking route
+                      Toast.makeText(
+                              context, "Hike details not implemented yet", Toast.LENGTH_SHORT)
+                          .show()
+                    })
+              }
+            }
+          }
+        }
+      },
+      sheetPeekHeight = 400.dp) {}
```

Comment:
Question: Does this mean the max height that you can pull the collapsible menu? Is there a reason you hardcoded this value to 400 dp? It seems to me that either using a relative measure like 3/5 of the screen height or just completely not restricting the height might make more sense.

Reply:
The maximum height is : fill screen. The value of `400.dp` represents the _minimum_ height. I just played around with it and `400.dp` looked good, that's pretty much it.

Reply:
Oooh, 'peek' height, not 'peak' height, I'm sorry haha. In that case this should be fine.